### ISSUE TITLE
How to disable VCS versioning?

### ISSUE THREAD

**Issue by x2ADLcRoo33ZygUA** [composer/composer#11431]
I have a `helper/type` package with the following `composer.json`:

```JSON
{
	"name": "helper/type",
	"description": "First",
	"version": "1.0.2",
	"autoload": {
		"psr-4": {
			"Type\\": "src/"
		}
	}
}
```

And the `main/project` package with the following `composer.json`:

```JSON
{
	"name": "main/project",
	"description": "Second",
	"repositories": [
		{
			"type": "vcs",
			"url": "../helper/"
		}
	],
	"require": {
		"teaching/versions_inner": "1.0.2"
	}
}
```

When I try to `composer install` for `main/project` I get the following message:

```Text
Root composer.json requires helper/type 1.0.2, found helper/type[dev-master] but it does not match the constraint.
```

If I change `repositories[0].type` to `path`, then the installation succeeds.

From which I can conclude that manually specifying `version` in `composer.json` when using VCS is not only *undesirable*, but also **impossible**. Is it so? If so, can I change this behavior somehow? What if I want to use Git but don't want to use tags for version numbering?

**Comment by x2ADLcRoo33ZygUA** [comment]
After studying the question in more detail, I came to the following conclusions.
The key aspect in determining the version of a package is the **repository type**. Some (for example, *path*) form it based on data from `composer.json`. Others do it differently. For example, *svn* generates a list of versions based on tags and branches, and does not take into account information from `composer.json`.
In conclusion, there is no way to make Composer parse the `composer.json` file versions from the Git commits list and build a version list based on the data it contains, not a based on git tags and branches.
Can any of the more experienced Composer users confirm or refute my guesses?

**Comment by Seldaek** [comment]
Yes that is correct, VCS repos will read versions from tags/branches. You can however [alias it](https://getcomposer.org/doc/articles/aliases.md) by requiring `"dev-master as 1.0.2"` for ex.

**Comment by x2ADLcRoo33ZygUA** [comment]
@Seldaek Thank you.
